spec/requests/articles_spec.rb

```diff
@@ -1,42 +1,66 @@
-require 'rails_helper'
+# frozen_string_literal: true
 
-RSpec.describe "Articles", type: :controller do
+require 'rails_helper'
 
+RSpec.describe 'Articles', type: :controller do
+  # `include` done here in order to avoid
+  # namespace collision with RegistrationController
+  include FactoryBot::Syntax::Methods
   describe 'create' do
     it 'successfully creates a new article' do
-
-      usr = User.create(username: 'testuser',
-                        email: 'test@test.com',
-                        firstname: 'testuser',
-                        lastname: 'testuser', 
-                        password: 'password',
-                        password_confirmation: 'password')
-
-      article = Article.create(title: 'testarticle',
-                               body: 'test test test',
-                               user_id: usr.id)
-      
+      usr = create(:user)
+      sign_in usr
+      article = create(:article, user_id: usr.id, body: 'test test test')
       expect(Article.last.body).to eq('test test test')
-      
     end
   end
 end
 
 RSpec.describe 'ArticlesController', type: :controller do
-  
-  describe "GET index" do
-    it "has a 302 status code" do
+  describe 'GET index' do
+    it 'has a 302 status code' do
       @controller = ArticlesController.new
       get :index
       expect(response.status).to eq(302)
     end
   end
 
-  describe "GET new" do
-    it "has a 302 status code" do
+  describe 'GET new' do
+    it 'has a 302 status code' do
       @controller = ArticlesController.new
       get :new
       expect(response.status).to eq(302)
     end
   end
-end
\ No newline at end of file
+
+  include FactoryBot::Syntax::Methods
+  describe 'with valid params' do
+    it 'creates a new article' do
+      @controller = ArticlesController.new
+      usr = create(:user)
+      sign_in usr
+      expect do
+        # either this
+        # art = create(:article, user_id: usr.id)
+        # or this
+        post :create,
+             params: { article: { title: Faker::Book.title, body: Faker::Lorem.paragraph } }
+      end.to change(Article, :count).by(1)
```

Comment:
Nice one! I really like the `expect { something }.to { do something }` matchers, good work on that!

Just one nit-pick here, you could replace `change(Article, :count)` to `change { Article.count }`, they would do the same thing, so I won't insist on you doing this.

Reply:
Thank you.
I made the suggested replacement, it is functionally the same although `rubocop` complains about it.